docs/content/docs/ops/state/state_backends.md

```diff
@@ -325,6 +325,126 @@ public class MyOptionsFactory implements ConfigurableRocksDBOptionsFactory {
 
 {{< top >}}
 
+## Enabling Changelog
+
+// todo: Chinese version of all changed docs
+
+// todo: mention in [large state tuning]({{< ref "docs/ops/state/large_state_tuning" >}})? or 1.16?
+
+{{< hint warning >}} The feature is in experimental status. {{< /hint >}}
+
+{{< hint warning >}} Enabling Changelog may have a negative performance impact on your application (see below). {{< /hint >}}
+
+### Introduction
+
+Changelog is a feature that aims to decrease checkpointing time, and therefore end-to-end latency in exactly-once mode.
+
+Most commonly, checkpoint duration is affected by:
+
+1. Barrier travel time and alignment, addressed by
+   [Unaligned checkpoints]({{< ref "docs/ops/state/checkpointing_under_backpressure#unaligned-checkpoints" >}})
+   and [Buffer debloating]({{< ref "docs/ops/state/checkpointing_under_backpressure#buffer-debloating" >}})
+2. Snapshot creation time (so-called synchronous phase), addressed by Asynchronous snapshots
```

Comment:
I agree, it's confusing.

What I mean here is minimizing the synchronous phase duration by using some form of persistent data structures (CoW for Heap and immutable SSTables in RocksDB). 
This is [called](https://nightlies.apache.org/flink/flink-docs-release-1.14/docs/ops/state/state_backends/#the-embeddedrocksdbstatebackend) asynchronous checkpoints/snapshots.


WDYT about
```
2. Snapshot creation time (so-called synchronous phase), reduced by immutable (or asynchronous) snapshots. 
```
Or maybe just remove the clarification?

Reply:
That's what I am not sure about.

Snapshot creation time: (isn't that just preparing for changelogs to be uploaded)? 
And previously snapshot creat time: flush + prepare SSTs to be uploaded?

I think most of the time reduced is from the async phase (uploading before ahead, so the async phase does not take very long time).

Reply:
> Snapshot creation time: (isn't that just preparing for changelogs to be uploaded)?
> And previously snapshot creat time: flush + prepare SSTs to be uploaded?

Yes.

> Would these two parts significantly different?
> I think most of the time reduced is from the async phase

For changelog, it's usually under 100ms, without it can take tens of seconds (above p99%). That can sound not much, but this phase blocks the entire processing. I describe it [here](https://github.com/apache/flink/pull/18431#discussion_r790436952).

But actually this statement is about reduction by using Async checkpoints which were implemented long time ago, not related to the Changelog.